plugins/pulp_rpm/plugins/importers/yum/parse/rpm.py

```diff
@@ -143,3 +145,45 @@ def package_signature(headers):
         return None
     # gpg program uses the last 8 characters of the fingerprint
     return signature.split()[-1][-8:]
+
+
+def signature_enabled(config):
+    """
+    Check if the signature policy is enabled.
+
+    :param config: configuration instance passed to the importer
+    :type  config: pulp.plugins.config.PluginCallConfiguration
+
+    :return: true if enabled, false otherwise
+    :rtype: boolean
+    """
+
+    require_signature = config.get("require_signature", False)
+    allowed_keys = config.get("allowed_keys")
+    if require_signature or allowed_keys:
+        return True
+    return False
+
+
+def verify_signature(unit, config):
+    """
+    Verify package signature.
+
+    :param unit: model instance of the package
+    :type  unit: pulp_rpm.plugins.db.models.RPM/DRPM/SRPM
+
+    :param config: configuration instance passed to the importer
+    :type  config: pulp.plugins.config.PluginCallConfiguration
+
+    :raise: PulpCodedException if the signature check did not pass
+    """
+
+    signing_key = unit.signing_key
+    require_signature = config.get("require_signature", False)
+    allowed_keys = config.get("allowed_keys", [])
+    if require_signature and not signing_key:
+        raise PulpCodedException(error_code=error_codes.RPM1013, package=unit.filename)
+    if allowed_keys:
+        if signing_key and signing_key not in allowed_keys:
```

Comment:
the extracted signing key from the header will be always in lowercase, so i just called .lower() on  the incoming list of allowed_keys in case user provided the key in uppercase